src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1336,11 +1336,15 @@ def update_site_configs(cmd, resource_group_name, name, slot=None, number_of_wor
                         vnet_route_all_enabled=None,
                         generic_configurations=None):
     configs = get_site_configs(cmd, resource_group_name, name, slot)
+    app_settings = _generic_site_operation(cmd.cli_ctx, resource_group_name, name,
+                                           'list_application_settings', slot)
     if number_of_workers is not None:
         number_of_workers = validate_range_of_int_flag('--number-of-workers', number_of_workers, min_val=0, max_val=20)
     if linux_fx_version:
         if linux_fx_version.strip().lower().startswith('docker|'):
-            update_app_settings(cmd, resource_group_name, name, ["WEBSITES_ENABLE_APP_SERVICE_STORAGE=false"])
+            if ('WEBSITES_ENABLE_APP_SERVICE_STORAGE' not in app_settings.properties or
+                    app_settings.properties['WEBSITES_ENABLE_APP_SERVICE_STORAGE'] != 'true'):
```

Comment:
Why does this logic judge the `'true'` as string?

Reply:
@zhoxing-ms AppSettings are just a simple key value pair & the AppSettings Object definition in swagger doesn't use this a dict but a simple list of strings.